SD_Assignment1/src/main/java/ro/utcn/aron/persistence/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,409 @@
+
+		Map<String, Object> data2 = new HashMap<>();
+		String[] tags = question.getTags().split(",");
+		for (int i = 0; i < tags.length; i++) {
+			data2.put("questionid", questionId);
+			data2.put("tags", tags[i]);
+			insert2.execute(data2);
+		}
+
+		return questionId;
+	}
+
+	private void update(Question question) {
+		template.update("UPDATE question SET title = ?, body = ?, author = ?, creationdate = ? WHERE id = ? ",
+				question.getTitle(), question.getBody(), question.getAuthor(), question.getCreationDate(),
+				question.getId());
+	}
+
+	@Override
+	public List<Question> filterByTitle(String title) {
+		return template
+				.query("SELECT * FROM question",
+						(resultSet, i) -> new Question(resultSet.getInt("id"), resultSet.getString("title"),
+								resultSet.getString("body"), list2String(getTagsFromJdbc(resultSet.getInt("id"))),
+								resultSet.getString("author"), resultSet.getString("creationdate"),getAnswersFromJdbc(resultSet.getInt("id")), getQuestionScore(resultSet.getInt("id"))))
+				.stream().filter(q -> q.getTitle().contains(title)).collect(Collectors.toList());
+	}
+
+	@Override
+	public List<Question> filterByTag(String tag) {
+		return template.query(
+				"SELECT * FROM question JOIN tags ON question.id = tags.questionid WHERE tags.tags LIKE '%" + tag
+						+ "%'",
+				(resultSet, i) -> new Question(resultSet.getInt("id"), resultSet.getString("title"),
+						resultSet.getString("body"), list2String(getTagsFromJdbc(resultSet.getInt("id"))),
+						resultSet.getString("author"), resultSet.getString("author"),getAnswersFromJdbc(resultSet.getInt("id")), getQuestionScore(resultSet.getInt("id"))));
+	}
+
+	private List<String> getTagsFromJdbc(int questionId) {
+		return template.query("SELECT * FROM tags WHERE questionid = ?",
+				(resultSet, i) -> new String(resultSet.getString("tags")), questionId);
+	}
+
+	private String list2String(List<String> tagList) {
+		String text = "";
+		for (int i = 0; i < tagList.size() - 1; i++) {
+			text += tagList.get(i) + ",";
+		}
+		text += tagList.get(tagList.size() - 1);
+
+		return text;
+	}
+
+	@Override
+	public void answerQuestion(int questionid, String user, String answer) {
+		SimpleJdbcInsert insert = new SimpleJdbcInsert(template);
+		insert.setTableName("answers");
+		insert.setGeneratedKeyName("answerid");
+
+		Map<String, Object> data = new HashMap<>();
+		data.put("questionid", questionid);
+		data.put("author", user);
+		data.put("answer", answer);
+
+		data.put("creationdate", new Date().toString());
+		insert.execute(data);
+	}
+
+	private List<Answer> getAnswersFromJdbc(int questionid) {
+		return template.query("SELECT * FROM answers WHERE questionid = ?",
+				(resultSet, i) -> new Answer(resultSet.getInt("answerid"),resultSet.getString("answer"), resultSet.getString("author"),
+						resultSet.getString("creationdate"), getAnswerScore(resultSet.getInt("answerid")) ),
+				questionid);
+	}
+	
+	private int getAnswerScore(int answerid) {
```

Comment:
These two methods for calculating scores should have been in the business logic layer (in your services). The database interface should have simply exposed a `List<Vote> findAnswerVotes(Answer a)` method, which would be called by the service and then used to calculate the scores.